### DISCUSSION TITLE
Alpinejs: change values inside children from parent element

### DISCUSSION THREAD

**Discussion by emmatraversy** [alpinejs/alpine#3180] [1. Help]
I have some nested custom select menus by alpinejs:

```
<template x-for="(attribute, rootindex) in attributes">
    <div x-ref="attribute.name"
     x-data="{open: false, selectedIndex: false, activeIndex: null, selectedItem:false, items:[...], placeholder:'Choose '+attribute.label }"
     class="my-5">
     ...
```
you can see each child has its own states/variables like open, selectedIndex, selectedItem, ... in a situations I want to set some of these states:
I tried using x-ref on childs but it wont work and it will return undefined! this function is at parent level:
```
    deleteAttribute(index) {
      for (let i = index; i < this.attributes.length; i++) {
        delete this.selected[this.attributes[i].name];
        this.$refs[this.attributes[i].name].selectedItem = false;
      }
```
Could you please tell me how can access children states? for example I want to set some of the 'selectedItem' to false in specific children

Many thanks in advance

**Comment by ekwoka** [comment]
Refs only go to the element not the component.


There can be a lot of different ways to accomplish depending on the specific needs. But mostly it involves the children accessing a prop on the parent.

**Reply by emmatraversy** [reply]
I created arrays for activeIndex, selectedIndex, and selectedItem in the parent and it works fine now but I was wondering if there is a way to change a property of children from parent? could you shed some light on this please?
Thanks
